tests/precice-adapter-config.json

```
@@ -3,8 +3,8 @@
   "config_file_name": "dummy.xml",
   "interfaces": {
     "Dummy-Mesh": {
-      "write_data_name": "Dummy-Write",
-      "read_data_name": "Dummy-Read",
+      "write_data_names": ["Dummy-Write"],
+      "read_data_names": ["Dummy-Read"],
```

```suggestion
      "write_data": ["Dummy-Write"],
      "read_data": ["Dummy-Read"],
```
I think this is a topic for another PR but I think we could just write `read_data`/`write_data` and then have a list of one or more (unique) names. I don't think we really need the suffix `_names`. Can you open an issue?

As far as I know `json` does not support unordered lists of unique objects ( = sets ). So we have to go with lists. In this case it would be reasonable to check the user input for duplicated names. If a duplicate occurs we should raise an error/warning.